commit 57e349b1b0df1aee2dcd19dae1f324bde25ff8f0
Author: H.J. Lu <hjl.tools@gmail.com>
Date:   Wed Dec 8 07:02:27 2021 -0800

    Disable DT_RUNPATH on NSS tests [BZ #28455]
    
    The glibc internal NSS functions should always load NSS modules from
    the system.  For testing purpose, disable DT_RUNPATH on NSS tests so
    that the glibc internal NSS functions can load testing NSS modules
    via DT_RPATH.
    
    This partially fixes BZ #28455.
    
    Reviewed-by: Carlos O'Donell <carlos@redhat.com>

Conflicts:
	nss/Makefile
	  (different test backport order)

diff --git a/nss/Makefile b/nss/Makefile
index 18a27d79d26f094a..648adf4bb7b1ee6a 100644
--- a/nss/Makefile
+++ b/nss/Makefile
@@ -214,3 +214,13 @@ endif
 
 $(objpfx)tst-nss-files-alias-leak.out: $(objpfx)/libnss_files.so
 $(objpfx)tst-nss-files-alias-truncated.out: $(objpfx)/libnss_files.so
+
+# Disable DT_RUNPATH on NSS tests so that the glibc internal NSS
+# functions can load testing NSS modules via DT_RPATH.
+LDFLAGS-tst-nss-test1 = -Wl,--disable-new-dtags
+LDFLAGS-tst-nss-test2 = -Wl,--disable-new-dtags
+LDFLAGS-tst-nss-test3 = -Wl,--disable-new-dtags
+LDFLAGS-tst-nss-test4 = -Wl,--disable-new-dtags
+LDFLAGS-tst-nss-test5 = -Wl,--disable-new-dtags
+LDFLAGS-tst-nss-test_errno = -Wl,--disable-new-dtags
+LDFLAGS-tst-nss-test_gai_hv2_canonname = -Wl,--disable-new-dtags
